### ISSUE TITLE
Request for integration - ESPHome Light Component - Additional state that accepts hex values

### ISSUE THREAD

**Issue by bigd-73** [iobroker-community-adapters/ioBroker.esphome#13]
Hello Dutchman,

an additional state that accepts hex values would be ideal, e.g. # 00ff00. A function in the adapter breaks down the value into the RGB values and forwards them to ESPHome. So far I've always solved this externally with a script. 

```
function hexToRgb(hex) {
  // Expand shorthand form (e.g. "03F") to full form (e.g. "0033FF")
  var shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
  hex = hex.replace(shorthandRegex, function(m, r, g, b) {
    return r + r + g + g + b + b;
  });

  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  return result ? {
    r: parseInt(result[1], 16),
    g: parseInt(result[2], 16),
    b: parseInt(result[3], 16)
  } : null;
}

return (hexToRgb(hex).r) + "," + (hexToRgb(hex).g) + "," + (hexToRgb(hex).b);
```

Thanks
Mario

**Comment by DutchmanNL** [comment]
not provided within the API, but will build the function HEX is very useful for certain situation agree !

**Comment by DutchmanNL** [comment]
implemented in >= 0.1.4

```
function hexToRgb(hex) {
    const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
    hex = hex.replace(shorthandRegex, function (m, r, g, b) {
        return r + r + g + g + b + b;
    });
 
    const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
    return result ? {
        red: parseInt(result[1], 16),
        green: parseInt(result[2], 16),
        blue: parseInt(result[3], 16)
    } : null;
}
 
function rgbToHex(r, g, b) {
    function componentToHex(color) {
        const hex = color.toString(16);
        return hex.length == 1 ? '0' + hex : hex;
    }
    const hex = '#' + componentToHex(r) + componentToHex(g) + componentToHex(b);
    return hex;
}
 
const rgb = hexToRgb(`ff4000`) 
console.log(rgb.red)
 
const hex = rgbToHex(rgb.red,rgb.blue,rgb.green) 
console.log(hex)
```
